gnat2goto/driver/range_check.adb

```diff
@@ -267,12 +264,51 @@ package body Range_Check is
       Value_Expr_Type : constant Irep :=
         Follow_Symbol_Type (Get_Type (Value_Expr), Global_Symbol_Table);
 
-      Op_Geq : constant Irep := New_Irep (I_Op_Geq);
-      Op_Leq : constant Irep := New_Irep (I_Op_Leq);
+      type Adjusted_Value_And_Bounds_T is
+        record
+           Value_Expr : Irep;
+           Upper_Bound : Irep;
+           Lower_Bound : Irep;
+        end record;
+
+      function Get_Adjusted_Value_And_Bounds
+         return Adjusted_Value_And_Bounds_T;
+      function Get_Adjusted_Value_And_Bounds
+         return Adjusted_Value_And_Bounds_T
+      is
+         Greater_Width : constant Boolean :=
+          Get_Width (Bound_Type) > Get_Width (Value_Expr_Type);
```

Comment:
:pick: In this case I think it'd actually look better with the condition being inlined